subprojects/shipkit/src/main/groovy/org/shipkit/internal/comparison/DependencyInfoComparator.java

```diff
@@ -0,0 +1,53 @@
+package org.shipkit.internal.comparison;
+
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.shipkit.internal.comparison.diff.Diff;
+import org.shipkit.internal.comparison.diff.FileDiffGenerator;
+import org.shipkit.internal.gradle.java.tasks.ComparePublications;
+
+import java.io.File;
+
+import static org.shipkit.internal.util.ArgumentValidation.notNull;
+
+public class DependencyInfoComparator {
```

Comment:
I gave some feedback about this guy in the PR comments. Perhaps we can simplify the implementation / make it more generic?

Reply:
I made it StringComparator, and moved all the complexity from this class (also from ComparePublications) regarding formatting the output to a new class ComparePublicationsResultFormatter.